src/DeFiScripts.sol

```diff
@@ -327,3 +330,111 @@ contract ApproveAndSwap {
         IERC20(sellToken).forceApprove(to, 0);
     }
 }
+
+// === Morpho actiosn ===
+contract MorphoVaultActions {
+    // To handle non-standard ERC20 tokens (i.e. USDT)
+    using SafeERC20 for IERC20;
+
+    function deposit(address vault, address asset, uint256 amount) external returns (uint256) {
+        IERC20(asset).forceApprove(vault, amount);
+        return IMetaMorpho(vault).deposit(amount, address(this));
+    }
+
+    function mint(address vault, address asset, uint256 shares) external returns (uint256 assets) {
+        IERC20(asset).forceApprove(vault, type(uint256).max);
+        assets = IMetaMorpho(vault).mint(shares, address(this));
+        IERC20(asset).forceApprove(vault, 0);
+    }
+
+    function withdraw(address vault, uint256 amount) external returns (uint256) {
+        return IMetaMorpho(vault).withdraw(amount, address(this), address(this));
+    }
+
+    function redeem(address vault, uint256 shares) external returns (uint256) {
+        return IMetaMorpho(vault).redeem(shares, address(this), address(this));
+    }
+}
+
+contract MorphoBlueActions {
+    // To handle non-standard ERC20 tokens (i.e. USDT)
+    using SafeERC20 for IERC20;
+
+    function borrow(
+        address morpho,
+        MarketParams memory marketParams,
+        uint256 assets,
+        uint256 shares,
+        address onBehalf,
+        address receiver
+    ) external returns (uint256, uint256) {
+        return IMorpho(morpho).borrow(marketParams, assets, shares, onBehalf, receiver);
+    }
+
+    function repay(
+        address morpho,
+        MarketParams memory marketParams,
+        uint256 assets,
+        uint256 shares,
+        address onBehalf,
+        bytes calldata data
+    ) external returns (uint256 assetsRepaid, uint256 sharesRepaid) {
+        if (assets > 0) {
+            IERC20(marketParams.loanToken).forceApprove(morpho, assets);
+            (assetsRepaid, sharesRepaid) = IMorpho(morpho).repay(marketParams, assets, shares, onBehalf, data);
+        } else {
+            IERC20(marketParams.loanToken).forceApprove(morpho, type(uint256).max);
+            (assetsRepaid, sharesRepaid) = IMorpho(morpho).repay(marketParams, assets, shares, onBehalf, data);
+            IERC20(marketParams.loanToken).forceApprove(morpho, 0);
+        }
+    }
+
+    function supplyCollateral(
```

Comment:
Similar to our Comet functions, we probably want a way to `Supply collateral and borrow` in a Quark script.  Same with `Repay and withdraw collateral`.

Reply:
oh yeah haha.
There was a moment I wanted to put this PR with another PR as one. 
These two actions got mistakenly made into the second PR 😓 instead of this one.
I think I will just move that changes to here, and remove that from the second PR then.